### ISSUE TITLE
UnknownError when pairing device

### ISSUE THREAD

**Issue by SoftwareRat** [jkcoxson/netmuxd#5]
OS: Fedora 35
iPhone: iPhone XS Max
Using latest iOS 15.4.1

```
❯ sudo ./netmuxd --disable-unix --host 127.0.0.1
[sudo] password for softwarerat: 
Starting netmuxd
Starting mDNS discovery
Listening on 127.0.0.1:27015
WARNING: Running in host mode will not work unless you are running a daemon in unix mode as well
Adding device 00008020-0004791A1446002E
[2022-04-21T17:40:14Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError
Adding device 00008020-0004791A1446002E
[2022-04-21T17:40:15Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError
Adding device 00008020-0004791A1446002E
[2022-04-21T17:40:16Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError
thread 'tokio-runtime-worker' panicked at 'range end index 421 out of range for slice of length 16', src/raw_packet.rs:62:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Adding device 00008020-0004791A1446002E
[2022-04-21T17:43:12Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError
Adding device 00008020-0004791A1446002E
[2022-04-21T17:44:02Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError
Adding device 00008020-0004791A1446002E
[2022-04-21T17:44:22Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError
Adding device 00008020-0004791A1446002E
[2022-04-21T17:44:32Z ERROR netmuxd::heartbeat] Failed to create heartbeat client for udid 00008020-0004791A1446002E: UnknownError

```

**Comment by jkcoxson** [comment]
Do you have usbmuxd running as well?

**Comment by SoftwareRat** [comment]
> Do you have usbmuxd running as well?

Yes, I am using the script from https://github.com/powenn/AltServer-Linux-ShellScript which automate the process

**Comment by jkcoxson** [comment]
Can you run it with logs? 
``sudo RUST_LOG=trace ./netmuxd —disable-unix —host 127.0.0.1``

**Comment by jkcoxson** [comment]
Also, looking into AltServer-Linux_ShellScript, it does not automatically start usbmuxd, it is up to the user. Ensure that it is running.

**Comment by SoftwareRat** [comment]
Here: https://gist.github.com/SoftwareRat/6fd25713f01019d258659ceacffea22a

**Comment by jkcoxson** [comment]
Looks like that log was successfully

**Comment by SoftwareRat** [comment]
AltStore says „unable to find AltServer“

**Comment by jkcoxson** [comment]
That’s a problem with AltServer then. The only connection netmuxd manages is AltServer to device, not the other way around.

**Comment by SoftwareRat** [comment]
But when I wait long enough it says „UnknownError“ thing

**Comment by adityasanghvi** [comment]
@SoftwareRat Follow these steps, it should help -

1. Kill all usbmuxd and netmuxd instances 
2. Connect your idevice to the linux machine on USB
3. Start usbmuxd as root
4. Run AltServer and refresh any sideloaded ipa
5. Remove the phone from USB and bring it on the same wi-fi network as the linux machine. 
6. Kill AltServer instance
7. Run netmuxd as root and see that it finally adds this idevice

These steps helped me resolve this "UnknownError".

**Comment by superrnovae** [comment]
Steps from above comment and the instructions from the README will get things working, but you will have to force start usbmuxd at every single boot before you start the netmuxd daemon. At first I've done that by specifying the requirement in netmuxd systemd service file, but later realized I don't need the usbmuxd at all. I now start netmuxd without any arguments, and let it create the usbmuxd socket, I also launch AltServer without the environment variable and it connects to the socket created by netmuxd without any issues. All my apps are being successfully refreshed.

**Comment by ramondunker** [comment]
Unfortunately this didn't help for me. I tried it on a RPI and on a x64 machine. On both devices I'm seeing the same error. I followed the exact steps in the README file but can't get it to work. Wired mode works flawlessly though.

**Comment by SoftwareRat** [comment]
> @SoftwareRat Follow these steps, it should help -
> 
>     1. Kill all usbmuxd and netmuxd instances
> 
>     2. Connect your idevice to the linux machine on USB
> 
>     3. Start usbmuxd as root
> 
>     4. Run AltServer and refresh any sideloaded ipa
> 
>     5. Remove the phone from USB and bring it on the same wi-fi network as the linux machine.
> 
>     6. Kill AltServer instance
> 
>     7. Run netmuxd as root and see that it finally adds this idevice
> 
> 
> These steps helped me resolve this "UnknownError".

Doesn't work for me

**Comment by Nosskirneh** [comment]
I have the same problem, and I think I've run out of options to try. Normal usbmuxd pairing works, as soon as I try to run netmuxd (with or without usbmuxd running in the background), it just doesn't find the device (`sudo idevice_id --list`). Any ideas?

**Comment by jkcoxson** [comment]
Not without logs or other info

**Comment by Nosskirneh** [comment]
Commands and logs: [commands.txt](https://github.com/jkcoxson/netmuxd/files/9737851/commands.txt) – I've only replaced my UDID.

iPhone 13 Pro, iOS 16.0.2. Raspberry Pi 3B running:
```
~ uname -a
Linux raspberrypi 5.15.32-v7+ #1538 SMP Thu Mar 31 19:38:48 BST 2022 armv7l GNU/Linux
```

Any other information that would be needed?

**Comment by Macleykun** [comment]
heya, also having this issue after installing altstore on my device
if there's anything i can help with, just ask!

**Comment by Nosskirneh** [comment]
The latest release fixed this problem for me. Now I get an entry saying `Adding device <UDID>`, which I didn't before. Very appreciated update, thanks @jkcoxson!

**Comment by wkddudwo** [comment]
I solved the problem.
./netmuxd --disable-unix --host 127.0.0.1
Instead of
RUST_LOG=trace ./netmuxd —disable-unix —host 127.0.0.1
If you run altserver after executing the log check command with , it works normally.

**Comment by yoonsikp** [comment]
I'm also having this issue, I keep getting an UnknownError, but USB backups work fine

**Comment by jkcoxson** [comment]
Please ensure this is still an issue with the latest commits.

**Comment by yoonsikp** [comment]
I think it's fixed
